src/HybridApp/Services/ProductImageUrlProvider.cs

```diff
@@ -0,0 +1,9 @@
+﻿using eShop.WebAppComponents.Services;
+
+namespace eShop.HybridApp.Services;
+
+public class ProductImageUrlProvider : IProductImageUrlProvider
+{
+    public string GetProductImageUrl(int productId) 
+        => $"{MauiProgram.MobileBffCatalogBaseUrl}api/v1/catalog/items/{productId}/pic";
```

Comment:
This was even easier than I thought: the Mobile BFF already proxies the entire catalog API, so I just had to use the correct URL, and the image API was already available.

Reply:
> This was even easier than I thought: the Mobile BFF already proxies the entire catalog API, so I just had to use the correct URL, and the image API was already available.

Surely it should not do that. The point of Mobile BFF is to publicly expose some endpoints that would not normally be publicly reachable from the internet. If it proxies everything, there's no point even having a Mobile BFF and the app owner might as well set up a public HTTP ingress for CatalogAPI.

Reply:
I think right now it seems a bit odd because the Hybrid App is a read-only view of _only_ the catalog and is not yet a full experience. In the future it would likely need to proxy multiple different services (cart, orders, etc.), plus also have some bespoke mobile-specific APIs (maybe tracking some app preferences/settings, usage insights/telemetry, etc.). But the whole HybridApp is really just 2 pages now (catalog list + item details), so the pattern isn't apparent.

It's probably reasonable to argue that instead it should not proxy _any_ entire service, and instead only specific APIs the app needs.

If you or anyone has thoughts on this I'd be happy to adjust to the pattern. I'm not quite sure what principles we are applying here, so I'd like to follow whatever we recommend.

Reply:
I'll give this a try. Given how simple the app is, it's _probably_ not that hard to do.

Reply:
Alright, it wasn't entirely trivial, but also not *too* complicated. It's a bit ugly, though. The challenge is that I had to add a specific mapping for each catalog-api that the mobile-bff needed to expose to the CatalogService (which is shared between WebApp and HybridApp). I kept the exact same API pattern that catalog-api already had because it means we can share the same CatalogService implementation. That's what a bit yucky: the way we have CatalogService right now *requires* that the exact same catalog APIs are exposed in exactly the same way (except the URL base address). Otherwise we'd need an `ICatalogService` and two different implementations, which would also be a bit yucky.

Reply:
That's great - this looks very good now! I don't even think it's ugly TBH - it feels correct to specify each endpoint that you want to expose publicly.

Reply:
Beauty is in the eye of the BFF.